Refactor LengthMode handling in SVGLengthContext

Ahmad-S792 opened the pull request:
#### fdbaf754005cabddd3b928228f2bc6dfdbae88ab
<pre>
Refactor LengthMode handling in SVGLengthContext

<a href="https://bugs.webkit.org/show_bug.cgi?id=279707">https://bugs.webkit.org/show_bug.cgi?id=279707</a>
<a href="https://rdar.apple.com/135995385">rdar://135995385</a>

Reviewed by Alan Baradlay.

Merge: <a href="https://chromium.googlesource.com/chromium/src.git/+/1368ac8914b34af0187b55e048477fd8abf191a2">https://chromium.googlesource.com/chromium/src.git/+/1368ac8914b34af0187b55e048477fd8abf191a2</a>

This patch introduces new helper function `dimensionForLengthMode`, which
selects the correct &quot;dimension&quot; factor to scale and with this function,
we would be able to reduce three repeated `switch` cases.

* Source/WebCore/svg/SVGLengthContext.cpp:
(WebCore::dimensionForLengthMode):
(WebCore::SVGLengthContext::valueForLength):
(WebCore::SVGLengthContext::convertValueFromUserUnitsToPercentage const):
(WebCore::SVGLengthContext::convertValueFromPercentageToUserUnits const):

Canonical link: <a href="https://commits.webkit.org/283671@main">https://commits.webkit.org/283671@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3a9ccd1a8daa9b447979f849dc2cea95b9daa80b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/67002 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/46377 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/19624 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/71037 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/18135 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/69120 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/54176 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/17918 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/53735 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/12201 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/70069 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/42654 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/57970 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/34255 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/39325 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/15364 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/16489 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/61245 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/15705 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/72738 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/10959 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/15041 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/61206 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/10991 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/58029 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/61281 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14799 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/8998 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/2605 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/42184 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/43261 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/44444 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/43004 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alanbaradlay commented on Source/WebCore/svg/SVGLengthContext.cpp:
```diff
@@ -88,6 +88,20 @@ float SVGLengthContext::resolveLength(const SVGElement* context, SVGUnitTypes::S
     return x.valueAsPercentage();
 }
 
+static inline float dimensionForLengthMode(SVGLengthMode mode, std::optional<FloatSize> viewportSize)
```
This should just be FloatSize viewportSize.

Ahmad-S792 replied:
I tried `const FloatSize&` but it gives compile error.

alanbaradlay replied:
FloatSize should be sufficient, but you need to update 2 of the callsites too (*viewportSize).

Ahmad-S792 replied:
On it. Testing locally.

Ahmad-S792 replied:
Done and pushed update with feedback. :-)

Ahmad-S792 committed:
Refactor LengthMode handling in SVGLengthContext

https://bugs.webkit.org/show_bug.cgi?id=279707
rdar://135995385

Reviewed by Alan Baradlay.

Merge: https://chromium.googlesource.com/chromium/src.git/+/1368ac8914b34af0187b55e048477fd8abf191a2

This patch introduces new helper function `dimensionForLengthMode`, which
selects the correct "dimension" factor to scale and with this function,
we would be able to reduce three repeated `switch` cases.

* Source/WebCore/svg/SVGLengthContext.cpp:
(WebCore::dimensionForLengthMode):
(WebCore::SVGLengthContext::valueForLength):
(WebCore::SVGLengthContext::convertValueFromUserUnitsToPercentage const):
(WebCore::SVGLengthContext::convertValueFromPercentageToUserUnits const):

Canonical link: https://commits.webkit.org/283671@main
